packages/recommend-core/src/getBatchRecommendations.ts

```diff
@@ -28,23 +30,73 @@ export type GetBatchRecommendations<TObject> = {
   keys: BatchKeyPair[];
   queries: Array<BatchQuery<TObject>>;
   recommendClient: RecommendClient;
-};
+} & Personalization;
 
 export type BatchRecommendations<TObject> = {
   recommendations: Array<ProductRecord<TObject>>;
   trendingFacets: TrendingFacetHit[];
 };
 
+const isTrendingFacetsQuery = <TObject>(
+  query: BatchQuery<TObject>
+): query is TrendingQuery => {
+  return query.model === 'trending-facets';
+};
+
+const isRecommendedForYouQuery = <TObject>(
+  query: BatchQuery<TObject>
+): query is RecommendedForYouQuery => {
+  return query.model === 'recommended-for-you';
+};
+
 export async function getBatchRecommendations<TObject>({
   keys,
   queries,
   recommendClient,
+  region,
+  userToken,
 }: GetBatchRecommendations<TObject>): Promise<
   Record<string, BatchRecommendations<TObject>>
 > {
   recommendClient.addAlgoliaAgent('recommend-core', version);
 
-  const response = await recommendClient.getRecommendations<TObject>(queries);
+  let _queries = queries;
+
+  /**
+   * Big block of duplicated code, but it is fine since it is experimental and will be ported to the API eventually.
+   * This is a temporary solution to get recommended personalization.
+   */
+  if (region && userToken) {
+    recommendClient.addAlgoliaAgent('personalization');
```

Comment:
We're tracking the experimentation here. (Needs to be updated in the other files too.)

```suggestion
    recommendClient.addAlgoliaAgent('experimental-personalization');
```
Or:

```suggestion
    recommendClient.addAlgoliaAgent('frontend-personalization');
```

Reply:
i like `experimental-personalization` 👍